### PR TITLE
ITL-1

### PR DESCRIPTION
**Pull request by UltraMKF12**

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -1,34 +1,9 @@
-<script setup lang="ts"></script>
+<script setup lang="ts">
+import TodoHeader from './components/TodoHeader.vue'
```
Comment (zsolt-wbg) [inline src/App.vue]: Please add this code block in tsconfig.json in order to use import paths with '@/...' 
`"paths": {
            "@/*": [
                "src/*"
            ]
        },`

File: src/App.vue [hunk 2]
```diff
@@ -1,34 +1,9 @@
-<script setup lang="ts"></script>
+<script setup lang="ts">
```
Comment (zsolt-wbg) [inline src/App.vue]: The order in a vue document should be: <template>, <script> and <style>

File: src/components/TodoHeader.vue [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<script setup lang="ts">
```
Comment (zsolt-wbg) [inline src/components/TodoHeader.vue]: Same here

File: src/components/TodoHeader.vue [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+import PlusSymbol from './icons/plus_icon.vue'
+defineEmits(['addTodo'])
+</script>
+
+<template>
+  <div class="flex justify-between">
+    <h1 class="text-7xl font-bold">To do List</h1>
+    <button @click="$emit('addTodo')">
```
Comment (zsolt-wbg) [inline src/components/TodoHeader.vue]: Create Base button component if you intend to use it more than one time.

File: src/components/icons/plus_icon.vue [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<template>
```
Comment (zsolt-wbg) [inline src/components/icons/plus_icon.vue]: Please name vue components with PascalCase and ts files with kebab-case

File: src/components/icons/plus_icon.vue [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  <svg width="60" height="60" viewBox="0 0 60 60" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (zsolt-wbg) [inline src/components/icons/plus_icon.vue]: Please add HeroIcons to the project in order to use icons via the projects
https://heroicons.com/

**Review by zsolt-wbg — CHANGES_REQUESTED** [review]

File: src/components/icons/PlusIcon.vue [hunk 7]
```diff
@@ -0,0 +1,13 @@
+<template>
```
Comment (szabolcs3) [inline src/components/icons/PlusIcon.vue]: You can delete this file.

File: src/components/TodoHeader.vue [hunk 8]
```diff
@@ -0,0 +1,14 @@
+<template>
+  <div class="flex justify-between">
+    <h1 class="text-7xl font-bold">To do List</h1>
+    <BaseButton>
```
Comment (szabolcs3) [inline src/components/TodoHeader.vue]: You don't need to wrap around the icon with button

File: src/components/BaseButton.vue [hunk 9]
```diff
@@ -0,0 +1,5 @@
+<template>
+  <slot></slot>
```
Comment (szabolcs3) [inline src/components/BaseButton.vue]: Please use `<button>` html tag here and add some attributes to it so you can easily change the button attributes.
```
Example: 

- Props (label, disabled, text color, background color, button text)
- Emits (click)
 ```

**Review by szabolcs3 — CHANGES_REQUESTED** [review]

File: src/components/TodoHeader.vue [hunk 10]
```diff
@@ -0,0 +1,11 @@
+<template>
+  <div class="flex justify-between">
+    <h1 class="text-7xl font-bold">To do List</h1>
+    <PlusCircleIcon @click="$emit('addTodo')" class="h-16 w-16 text-green-400 cursor-pointer" />
```
Comment (zsolt-wbg) [inline src/components/TodoHeader.vue]: Please use the correct attribute order! Here is an example: https://github.com/vuejs/eslint-plugin-vue/blob/master/docs/rules/attributes-order.md
But we will send you the "Vue Bible" :)

Comment (zsolt-wbg) [inline src/components/TodoHeader.vue]: Please use a separate function for emit, for instance:
`@click="addTodo"`
`function addTodo() {
   emit('addTodo')
}`

File: src/components/BaseButton.vue [hunk 11]
```diff
@@ -0,0 +1,16 @@
+<template>
+  <button></button>
+</template>
+
+<script setup lang="ts">
+interface Props {
```
Comment (zsolt-wbg) [inline src/components/BaseButton.vue:11]: Please add the props directly to the button element tag, like this:
```
<button
   :label="label"
   :disabled="disabled"
   :class="[textColor, backgroundColor]"
>
      { buttonText }
</button>
```

**Review by zsolt-wbg — CHANGES_REQUESTED** [review]

**Review by szabolcs3 — APPROVED** [review]

**Review by zsolt-wbg — APPROVED** [review]

**Pull request merged**